parser_library/src/parsing/grammar/hlasmparser_multiline.g4

```diff
@@ -0,0 +1,581 @@
+				disable_continuation();
+			}
+			|
+			{
+				if (!$pending_empty_op)
+					throw NoViableAltException(this);
+			}
+			ca_op=ca_op_expr
+			{
+				$pending_empty_op = false;
+			}
+			{
+                if ($ca_op.op)
+                    $operands.push_back(std::move($ca_op.op));
+                else
+                    $operands.push_back(std::make_unique<semantics::empty_operand>(provider.get_empty_range($ca_op.start)));
+			}
+		)*
+		EOF
+		{
+			if ($pending_empty_op)
+				$operands.push_back(std::make_unique<semantics::empty_operand>(provider.get_empty_range(_input->LT(-1))));
+		}
+	);
+	finally
+	{
+		disable_continuation();
+		if ($first_token)
+			collector.set_operand_remark_field(std::move($operands), std::move($remarks), provider.get_range($first_token, _input->LT(-1)));
+	}
+op_rem_body_ca_var_def locals [bool pending_empty_op = false, std::vector<range> remarks, std::vector<operand_ptr> operands, antlr4::Token* first_token = nullptr]
+	:
+	EOF
+	{
+		collector.set_operand_remark_field(provider.get_range(_localctx));
+	}
+	|
+	SPACE+
+	(
+		EOF
+		{
+			collector.set_operand_remark_field(provider.get_range(_localctx));
+		}
+		|
+		{
+			$first_token = _input->LT(1);
+		}
+		(
+			comma
+			{
+                $operands.push_back(std::make_unique<semantics::empty_operand>(provider.get_empty_range($comma.start)));
+				$pending_empty_op = true;
+			}
+			|
+			{
+				$pending_empty_op = false;
+			}
+			ca_op=ca_op_var_def
+			{
+                if ($ca_op.op)
+                    $operands.push_back(std::move($ca_op.op));
+                else
+                    $operands.push_back(std::make_unique<semantics::empty_operand>(provider.get_empty_range($ca_op.start)));
+			}
+		)
+		(
+			comma
+			{
+				if ($pending_empty_op)
+					$operands.push_back(std::make_unique<semantics::empty_operand>(provider.get_empty_range($comma.start)));
+				$pending_empty_op = true;
+			}
+			|
+			{
+				if (_input->LA(-1) == hlasmparser_multiline::COMMA)
+					enable_continuation();
+			}
+			(
+				SPACE
+				remark
+				{
+					$remarks.push_back(provider.get_range($remark.ctx));
+				}
+				CONTINUATION?
+				|
+				CONTINUATION
+			)
+			{
+				disable_continuation();
+			}
+			|
+			{
+				if (!$pending_empty_op)
+					throw NoViableAltException(this);
+			}
+			ca_op=ca_op_var_def
+			{
+				$pending_empty_op = false;
+			}
+			{
+                if ($ca_op.op)
+                    $operands.push_back(std::move($ca_op.op));
+                else
+                    $operands.push_back(std::make_unique<semantics::empty_operand>(provider.get_empty_range($ca_op.start)));
+			}
+		)*
+		EOF
+		{
+			if ($pending_empty_op)
+				$operands.push_back(std::make_unique<semantics::empty_operand>(provider.get_empty_range(_input->LT(-1))));
+		}
+	);
+	finally
+	{
+		disable_continuation();
+		if ($first_token)
+			collector.set_operand_remark_field(std::move($operands), std::move($remarks), provider.get_range($first_token, _input->LT(-1)));
+	}
```

Comment:
Is it possible to have some common part that would get called for any of the three rules here since it is duplicated except for one line?

Something like:
```
op_rem_body_ca_var_def locals [bool pending_empty_op = false, std::vector<range> remarks, std::vector<operand_ptr> operands, antlr4::Token* first_token = nullptr]
	:
	EOF
	{
		collector.set_operand_remark_field(provider.get_range(_localctx));
	}
	|
	SPACE+
        rule_pre
	{
		if (!$pending_empty_op)
			throw NoViableAltException(this);
	}
	ca_op=ca_op_var_def
	{
		$pending_empty_op = false;
	}
	{
		if ($ca_op.op)
			$operands.push_back(std::move($ca_op.op));
		else
			$operands.push_back(std::make_unique<semantics::empty_operand>(provider.get_empty_range($ca_op.start)));
	}
        rule_post
        ...
```

Reply:
I am not sure I follow... the actions are the same because of the `ca_op` alias, however the rest are always different rules with 2 common alternatives.

Reply:
Alright, let's keep it